mailer.http.class.php

```diff
@@ -43,50 +56,64 @@ function sparkpostSend()
         $this->edebug('Response received');
 
         return $this->handle_response($result);
-
-
     }
 
+    /**
+     * Build the request body to be sent to the SparkPost API.
+     */
     protected function get_request_body()
     {
         $tracking_enabled = !!$this->options['enable_tracking'];
         $sender = $this->get_sender();
-        $body = array(
-            'recipients' => $this->get_recipients(),
-            'content' => array(
-                'from' => $sender,
-                'subject' => $this->Subject,
-                'headers' => $this->get_headers()
-            ),
-            'options' => array(
-                'open_tracking' => $tracking_enabled,
-                'click_tracking' => $tracking_enabled
-            )
+        $replyTo = $this->get_reply_to();
+        $body = array();
+
+        // add recipients
+        $body['recipients'] = $this->get_recipients();
+
+        // enable engagement tracking
+        $body['options'] = array(
+            'open_tracking' => $tracking_enabled,
+            'click_tracking' => $tracking_enabled
         );
 
+        // pass through either stored template or inline content
         if (!empty($this->options['template'])) {
-          $body['content']['template_id'] = $this->options['template'];
-          $body['substitution_data']['content'] = $this->Body;
-          $body['substitution_data']['subject'] = $this->Subject;
-          $body['substitution_data']['from_name'] = $sender['name'];
+            // stored template
+            $body['content']['template_id'] = $this->options['template'];
+
+            // supply substitution data so users can add variables to templates
+            $body['substitution_data']['content'] = $this->Body;
+            $body['substitution_data']['subject'] = $this->Subject;
+            $body['substitution_data']['from_name'] = $sender['name'];
+            $body['substitution_data']['from'] = $sender['name'] . ' <' . $sender['email'] . '>';
```

Comment:
Without looking at the plugin UI (which I'm about to do), is it obvious to users that these substitutions are available?

`<noid>` Also are we likely to hit a `substitution_data` size limit?  If the message contains say an article synopsis, it might be a sizable subs variable even with boilerplate held in the template. `</noid>`